### PR TITLE
Validate if bucket names are internal

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Validate if bucket names are internal

## Motivation and Context
This commit fixes a privilege escalation issue against
the S3 and web handlers. An authenticated IAM user
can:

- Read from or write to the internal '.minio.sys'
bucket by simply sending a properly signed
S3 GET or PUT request. Further, the user can
- Read from or write to the internal '.minio.sys'
bucket using the 'Upload'/'Download'/'DownloadZIP'
API by sending a "browser" request authenticated
with its JWT token.

## Regression
No

## How Has This Been Tested?
Handcrafted browser requests with JWT etc.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#95a9a9dd-8af8-4069-9e71-c765e8378562) |
| mint-dist-xl.sh | :x: [more...](#65f39892-23b4-4b2f-a1dd-57c52b3daef0) |
<a name="95a9a9dd-8af8-4069-9e71-c765e8378562"></a>
### 7476-b4ca9ce/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:30825
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 054bc8cb520a:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 53 seconds
(4/14) Running aws-sdk-ruby tests ... done in 20 seconds
(5/14) Running awscli tests ... done in 2 minutes and 12 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 41 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 21 seconds
(9/14) Running minio-go tests ... done in 2 minutes and 25 seconds
(10/14) Running minio-java tests ... done in 4 minutes and 36 seconds
(11/14) Running minio-js tests ... done in 48 seconds
(12/14) Running minio-py tests ... FAILED in 7 minutes and 42 seconds
{
  "status": "FAIL",
  "duration": 1301,
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "args": {
    "bucket_name": "minio-py-test-3fd80ac1-4c8c-414f-9a15-c6f28830dae6",
    "object_name": "e8405766-f1b7-480c-9b85-773d87941626"
  },
  "message": "AccessDenied: message: Access Denied",
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)"
}

Executed 11 out of 14 tests successfully.
```
<a name="65f39892-23b4-4b2f-a1dd-57c52b3daef0"></a>
### 7476-b4ca9ce/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:30707
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 28b58a9beea2:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-wfrd9kqygzv0wb",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3066,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:30707/aws-sdk-go-test-wfrd9kqygzv0wb: dial tcp 72.28.97.57:30707: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#30b8d422-49d6-4943-bd7a-3fbc5ccb0ee4) |
| mint-compression-dist-xl.sh | :x: [more...](#0ac4abfd-618b-4d0c-b2c8-010227630d28) |
| mint-compression-fs.sh | :x: [more...](#b14265eb-24e2-48d5-a3e0-d98fc8398790) |
| mint-fs.sh | :x: [more...](#5247ffec-3780-4752-a539-82660109ad07) |
<a name="0ac4abfd-618b-4d0c-b2c8-010227630d28"></a>
### 7476-839f6c0/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:31782
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3ec396b92f2e:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... FAILED in 21 minutes and 50 seconds
{
  "name": "awscli",
  "duration": 253703,
  "function": "aws --endpoint-url http://72.28.97.60:31782 s3 presign s3://awscli-mint-test-bucket-26760/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Checksum verification failed for downloaded object"
}

Executed 4 out of 14 tests successfully.
```
<a name="b14265eb-24e2-48d5-a3e0-d98fc8398790"></a>
### 7476-839f6c0/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:31443
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp edf0e9b3189c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-m1duqqsi4jwqbb",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 2865,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:31443/aws-sdk-go-test-m1duqqsi4jwqbb: dial tcp 72.28.97.57:31443: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="5247ffec-3780-4752-a539-82660109ad07"></a>
### 7476-839f6c0/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:30050
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9952fdce11d6:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-xbughy09nnczwh",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3074,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:30050/aws-sdk-go-test-xbughy09nnczwh: dial tcp 72.28.97.57:30050: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="30b8d422-49d6-4943-bd7a-3fbc5ccb0ee4"></a>
### 7476-839f6c0/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:31757
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8ddbe0d17493:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 2 minutes and 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-sshyaqbb6nf4v1",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 120153,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 159217D88382FF60, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

File: cmd/generic-handlers.go [hunk 1]
```diff
@@ -196,7 +196,9 @@ func guessIsBrowserReq(req *http.Request) bool {
 	if req == nil {
 		return false
 	}
-	return strings.Contains(req.Header.Get("User-Agent"), "Mozilla")
+	aType := getRequestAuthType(req)
+	return strings.Contains(req.Header.Get("User-Agent"), "Mozilla") && globalIsBrowserEnabled &&
+		(aType == authTypeJWT || aType == authTypeAnonymous)
 }
```
Comment (poornas) [inline cmd/generic-handlers.go:202]: can you update TestGuessIsBrowser

**Review by aead — DISMISSED** [review]
Tested against my "exploits". Fixes the issue AFAICS. LGTM.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#ea44c4ba-b60a-4319-a84b-07c645624f22) |
| mint-compression-fs.sh | :x: [more...](#7ed0afbe-eaa5-4bf6-a8f1-d4fb6af0ac1c) |
| mint-dist-xl.sh | :x: [more...](#28376725-5c91-4d6f-956a-3d4a877257bc) |
| mint-gateway-nas.sh | :x: [more...](#bfe514b5-d3e3-4f1b-b8e0-0eed4ac85dd9) |
<a name="7ed0afbe-eaa5-4bf6-a8f1-d4fb6af0ac1c"></a>
### 7476-94a7bb3/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:32598
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 63e0ea6d03e3:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 2 minutes and 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-gk3bn4wv364vt9",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 120290,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.60:32598/aws-sdk-go-test-gk3bn4wv364vt9: dial tcp 72.28.97.60:32598: i/o timeout",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="bfe514b5-d3e3-4f1b-b8e0-0eed4ac85dd9"></a>
### 7476-94a7bb3/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:30569
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0ac218de6cbf:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-c9dbqvjo2b6kdp",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 2714,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:30569/aws-sdk-go-test-c9dbqvjo2b6kdp: dial tcp 72.28.97.57:30569: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="28376725-5c91-4d6f-956a-3d4a877257bc"></a>
### 7476-94a7bb3/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:30703
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 47840e655fe4:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 4 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 50 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 0 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 29 seconds
(8/14) Running minio-dotnet tests ... done in 33 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 18 seconds
(10/14) Running minio-java tests ... done in 2 minutes and 34 seconds
(11/14) Running minio-js tests ... done in 40 seconds
(12/14) Running minio-py tests ... FAILED in 5 minutes and 1 seconds
{
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1266, in test_presigned_get_object_response_headers\n    returned_content_language = response.headers['Content-Language']\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/_collections.py\", line 152, in __getitem__\n    val = self._container[key.lower()]\nKeyError: 'content-language'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1888, in main\n    test_presigned_get_object_response_headers(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1274, in test_presigned_get_object_response_headers\n    raise Exception(err)\nException: 'content-language'\n",
  "status": "FAIL",
  "name": "minio-py:test_presigned_get_object_response_headers",
  "args": {
    "object_name": "266fc694-a069-4953-9f43-8aaa05f1955c",
    "bucket_name": "minio-py-test-0e630de1-a90d-4b7f-ac9f-4f5f7c30bb2c"
  },
  "message": "'content-language'",
  "duration": 139
}

Executed 11 out of 14 tests successfully.
```
<a name="ea44c4ba-b60a-4319-a84b-07c645624f22"></a>
### 7476-94a7bb3/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:31462
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a5f41f93f0d5:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-hi613d1z96bxfq",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3061,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:31462/aws-sdk-go-test-hi613d1z96bxfq: dial tcp 72.28.97.57:31462: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Comment by aead** [comment]
But the `TestGuessIsBrowser` test must be fixed. Currently it panics

**Review by poornas — DISMISSED** [review]
Tested by modifying browser requests - accessing objects in reserved bucket now gives a "specified bucket is not valid" error. LGTM

**Comment by poornas** [comment]
travis build failure needs to be fixed.

**Review by poornas — DISMISSED** [review]

**Review by aead — DISMISSED** [review]
Tested, LGTM. Also run the `TestGuessIsBrowser` locally.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#dcfa3bda-c44f-4d67-9643-5e76d015e544) |
| mint-worm.sh | :x: [more...](#e9bc69de-b42e-423c-be81-8b74675d4beb) |
<a name="dcfa3bda-c44f-4d67-9643-5e76d015e544"></a>
### 7476-a459aec/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:30006
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d0a85b765b8a:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 58 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 32 seconds
(8/14) Running minio-dotnet tests ... done in 35 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 23 seconds
(10/14) Running minio-java tests ... done in 4 minutes and 57 seconds
(11/14) Running minio-js tests ... done in 45 seconds
(12/14) Running minio-py tests ... FAILED in 8 minutes and 20 seconds
{
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "duration": 135,
  "args": {
    "object_name": "9016cd8f-a8a3-4106-9e5b-1c302262e8ce",
    "bucket_name": "minio-py-test-9ace1c03-38c0-40d5-be3d-0feea853bb32"
  },
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "message": "AccessDenied: message: Access Denied",
  "status": "FAIL"
}

Executed 11 out of 14 tests successfully.
```
<a name="e9bc69de-b42e-423c-be81-8b74675d4beb"></a>
### 7476-a459aec/mint-worm.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:31765
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            worm
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b5c6432d2c8a:/mint/log /tmp/mint-logs'

(1/1) Running worm tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "bucket-xqt3gts6saezppp10gc4lje64i7nh9gxwf6a6cs9lpqryo4fzhogcjh",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 3075,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:31765/bucket-xqt3gts6saezppp10gc4lje64i7nh9gxwf6a6cs9lpqryo4fzhogcjh: dial tcp 72.28.97.57:31765: connect: no route to host",
  "function": "PutAndDelete",
  "message": "WORM_MODE ON  - CreateBucket Failed",
  "name": "test worm mode",
  "status": "FAIL"
}

Executed 0 out of 1 tests successfully.
```

**Review by kannappanr — DISMISSED** [review]
LGTM

**Comment by kannappanr** [comment]
https://travis-ci.org/minio/minio/jobs/515460856#L2228 `TestWebCheckAuthorization` is failing

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-fs.sh | :x: [more...](#a5f780e9-59ac-4ca3-b4af-abc0866b6246) |
<a name="a5f780e9-59ac-4ca3-b4af-abc0866b6246"></a>
### 7476-ad149ad/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:32584
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b4a2b199f2b2:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-v3kc2rfb9yly5t",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3055,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:32584/aws-sdk-go-test-v3kc2rfb9yly5t: dial tcp 72.28.97.57:32584: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

**Review by kannappanr — APPROVED** [review]
LGTM

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#a0c2c0f5-b5ff-41d7-96e3-e63c86fa5395) |
| mint-fs.sh | :x: [more...](#265de94e-6212-4a67-9b38-53994ad7af26) |
| mint-dist-xl.sh | :x: [more...](#3d88f973-612f-4050-bd11-62edd060996a) |
<a name="265de94e-6212-4a67-9b38-53994ad7af26"></a>
### 7476-8f3cd03/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:32207
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp aa75f696c903:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-5l4fs1naojyjrh",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3059,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:32207/aws-sdk-go-test-5l4fs1naojyjrh: dial tcp 72.28.97.57:32207: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="3d88f973-612f-4050-bd11-62edd060996a"></a>
### 7476-8f3cd03/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:30939
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0657dfeb1463:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-m0kctfddlun2oh",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 1980,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:30939/aws-sdk-go-test-m0kctfddlun2oh: dial tcp 72.28.97.57:30939: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="a0c2c0f5-b5ff-41d7-96e3-e63c86fa5395"></a>
### 7476-8f3cd03/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:31240
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 52ede65df5d1:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-b1qgi4t0xlpb4r",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 2862,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:31240/aws-sdk-go-test-b1qgi4t0xlpb4r: dial tcp 72.28.97.57:31240: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Comment by kannappanr** [comment]
Merging as it has already been reviewed and approved by @aead and @poornas. New push is just a rebase.

**Pull request merged**